File: Java/PoliceStation.java
import java.util.ArrayList;
import java.util.List;

public class PoliceStation {
    private int idPoliceStation;
    private String policeStationName;
    private String policeStationAddress;
    private List<Police> listPolice;

    // Default constructor
    public PoliceStation() {
        this.idPoliceStation = 0;
        this.policeStationName = "";
        this.policeStationAddress = "";
        this.listPolice = new ArrayList<>();
    }

    // Parameterized constructor
    public PoliceStation(int idPoliceStation, String name, String address) {
        this.idPoliceStation = idPoliceStation;
        this.policeStationName = name;
        this.policeStationAddress = address;
        this.listPolice = new ArrayList<>();
    }

    // Getter methods
    public int getIdPoliceStation() {
        return this.idPoliceStation;
    }

    public String getPoliceStationName() {
        return this.policeStationName;
    }

    public String getPoliceStationAddress() {
        return this.policeStationAddress;
    }

    public List<Police> getListPolice() {
        return this.listPolice;
    }

    // Setter methods (ID tidak ada setter)
    public void setPoliceStationName(String name) {
        this.policeStationName = name;
    }

    public void setPoliceStationAddress(String address) {
        this.policeStationAddress = address;
    }

    public void setListPolice(List<Police> daftar) {
        this.listPolice = daftar;
    }

    // Tambah Police ke list
    public void addPolice(Police police) {
        this.listPolice.add(police);
    }

    // Display
    public void displayPoliceStation(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "Police Station ID : " + getIdPoliceStation());
        System.out.println(sp + "Name              : " + getPoliceStationName());
        System.out.println(sp + "Address           : " + getPoliceStationAddress());
        System.out.println();

        if (!listPolice.isEmpty()) {
            System.out.println(sp + "    Police Officers:");
            for (Police p : listPolice) {
                p.displayPolice(indent + 8); // tampilkan detail police
            }
        }
        System.out.println();
    }
}

File: Cpp/PoliceStation.cpp
#pragma once
#include<iostream>
#include<string>
#include<vector>
#include"Police.cpp"
using namespace std;

class PoliceStation {
private:
    int idPoliceStation;
    string policeStationName;
    string policeStationAddress;
    vector<Police> listPolice;

public:
    // Default constructor (ID auto-generate)
    PoliceStation() {
        this->idPoliceStation = 0;
        this->policeStationName = "";
        this->policeStationAddress = "";
    }

    // Parameterized constructor (ID auto-generate)
    PoliceStation(int idPoliceStation, string name, string address) {
        this->idPoliceStation = idPoliceStation;
        this->policeStationName = name;
        this->policeStationAddress = address;
    }

    // Getter methods
    int getIdPoliceStation() {
        return this->idPoliceStation;
    }

    string getPoliceStationName() {
        return this->policeStationName;
    }

    string getPoliceStationAddress() {
        return this->policeStationAddress;
    }

    vector<Police> getListPolice() {
        return this->listPolice;
    }

    // Setter methods (id tidak ada setter!)
    void setPoliceStationName(string name) {
        this->policeStationName = name;
    }

    void setPoliceStationAddress(string address) {
        this->policeStationAddress = address;
    }

    void setListPolice(vector<Police> daftar) {
        this->listPolice = daftar;
    }

    // Tambah Police ke list
    void addPolice(const Police& police) {
        this->listPolice.push_back(police);
    }

    void displayPoliceStation(int indent) {
        string sp(indent, ' ');
        cout << sp << "Police Station ID : " << getIdPoliceStation() << endl;
        cout << sp << "Name              : " << getPoliceStationName() << endl;
        cout << sp << "Address           : " << getPoliceStationAddress() << endl << endl;

        if (!listPolice.empty()) {
            cout << sp << "    Police Officers:" << endl;
            for (auto &p : listPolice) {
                p.displayPolice(indent + 8); // Tampilkan detail Police
            }
        }
        cout << endl;
    }

    // Destructor
    ~PoliceStation() {
    }
};
